pkg/chunked/compressor/compressor.go

```diff
@@ -202,7 +205,18 @@ func writeZstdChunkedStream(destFile io.Writer, outMetadata map[string]string, r
 	// total written so far.  Used to retrieve partial offsets in the file
 	dest := ioutils.NewWriteCounter(destFile)
 
-	tr := tar.NewReader(reader)
+	metaPackerBuffer := bytes.NewBuffer(nil)
```

Comment:
Would it make sense to inline the compression here, instead of building a large in-memory buffer only to compress it? Some layers are gigantic.